geode-pulse/src/integrationTest/java/org/apache/geode/tools/pulse/security/OauthSecurityConfigTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.tools.pulse.security;
+
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrlPattern;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.GenericXmlWebContextLoader;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(locations = {"classpath*:WEB-INF/pulse-servlet.xml"},
+    loader = GenericXmlWebContextLoader.class)
+@WebAppConfiguration
+@ActiveProfiles({"pulse.authentication.oauth"})
+public class OauthSecurityConfigTest {
+  @Autowired
+  private WebApplicationContext context;
+
+  private MockMvc mvc;
+
+  @Before
+  public void setup() {
+    mvc = MockMvcBuilders
+        .webAppContextSetup(context)
+        .apply(springSecurity())
+        .build();
+  }
+
+  @Test
+  public void redirectToOauth() throws Exception {
+    mvc.perform(get("/login.html")).andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrl(("http://localhost/oauth2/authorization/uaa")));
+
+    mvc.perform(get("http://localhost/oauth2/authorization/uaa"))
+        .andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrlPattern("http://xyz.com/uaa/oauth/**"));
```

Comment:
xyz.com is an actual domain name. Please use a test-specific domain name such as:

- `example.com` (`example.com` is intended mostly for documentation, but people use it for testing)
- `<any-subdomain>.test` (`.test` is a special top-level domain name designed for testing)
- `<any-subdomain>.example` (`.example` is another special top-level domain name)

Reply:
will do